Lib/typing.py

```diff
@@ -253,7 +253,8 @@ def _collect_parameters(args):
         if hasattr(t, '__typing_subst__'):
             if t not in parameters:
                 parameters.append(t)
-        else:
+        # We need to avoid bare Python types as those aren't really objects.
+        elif type(t) is not type:
```

Comment:
I liked your previous version of the comment ("... aren't really generic") better -- surely everything in Python is an object!

I'm not sure I follow how this check is correct. I'm guessing that the check is intended to disambiguate the case where a base class is e.g. `Base[S]` vs. plain `Base`, and the bug in the original code is that the author assumed that anything with a `__parameters__` attribute must be something of the form `X[Y]` where `X` is a class and `Y` is a type object; whereas plain `Base`, being itself generic, also has a `__parameters__` attribute. But couldn't there be other cases that we're still overlooking? The code before the bug was introduced checked for a few specific cases. Checking for `__parameters__` might feel more elegant, but is it more correct?

@serhiy-storchaka 

Finally: What if `type(t)` is a *subclass* of `type`? Shouldn't the check be `elif not isinstance(t, type):`?

Reply:
> I'm guessing that the check is intended to disambiguate the case where a base class is e.g. Base[S] vs. plain Base

Yep.

> might feel more elegant, but is it more correct?

Yeah that has me a little worried. I'm not foolish enough to say that this should cover all invalid cases :). I think restoring the old checks may be the safer option.